pkg/crypto/data_messages.go

```diff
@@ -24,7 +24,26 @@ import (
 	"go.thethings.network/lorawan-stack/v3/pkg/types"
 )
 
-func encryptMessage(key types.AES128Key, dir uint8, addr types.DevAddr, fCnt uint32, payload []byte, isFOpts bool) ([]byte, error) {
+type encryptionOptions struct {
+	frameIdentifier [4]byte
+}
+
+// EncryptionOption encapsulates custom LoRaWAN encryption options.
+type EncryptionOption func(*encryptionOptions)
+
+// WithFrameIdentifier specifies which bytes are to be used for
+// positions [1:5] during encryption.
+func WithFrameIdentifier(identifier [4]byte) EncryptionOption {
+	return func(encOpts *encryptionOptions) {
+		copy(encOpts.frameIdentifier[:], identifier[:])
+	}
+}
```

Comment:
Can you document this a bit more? What is the LoRaWAN aspect of this?

Reply:
This is a bit of poetic license from my side. The (updated) specification does not have names for the parts of the block:
![image](https://user-images.githubusercontent.com/36161392/169006956-2f274779-6b1c-4d16-8dcc-ee5f56193860.png)
The 4 bytes after the `0x01` are what I call the frame identifier. The naming itself comes from the erratum body:
![image](https://user-images.githubusercontent.com/36161392/169007328-0567bdcb-6f25-458e-98a5-4e14b04c9530.png)
They are called 'constants' there, but their role is to separate between different frame types - so maybe 'frame type block' is closer to what they really do ?

Reply:
Yes that sounds better to me.